### ISSUE TITLE
Can't open in photoshop or indesign images downloaded from instagram

### ISSUE THREAD

**Issue by nitriques** [shftcl/www-issues#30]
hello, i can't open in photoshop or indesign images downloaded from instagram. a problem with the analisys of data from the image appears.

Application version: 1.3.3
Client version: 5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36
Reported date: 7/11/16 6:05 PM
Reported by: A Google User

**Comment by munki-boy** [comment]
Change the file extension in the filename to .jpg or something like that.  Probably a jpeg file got named as .png or whatever.

**Comment by munki-boy** [comment]
Nah, doesn't work

**Comment by nitriques** [comment]
I still have to test, this bug comes from the chrome store lol.

But I think we do respect the original file extension.

**Comment by munki-boy** [comment]
Some of these sites send the wrong header, thought it might be that but not in this case.

**Comment by munki-boy** [comment]
The images aren't spec correct, Photoshop cares but camera raw/lightroom don't so you have to open via those into Photoshop or other Adobe apps. Other brand software is more tolerant.

**Comment by nitriques** [comment]
We parse images using C/C++ code to make sure they are valid and they pass validation. I wonder what's bothering photoshop.

**Comment by munki-boy** [comment]
I think Photoshop expects the kind of jpeg that has EXIF.  Both types start with the Jpeg magicbytes but only the first little bit. As soon as you pass them through something less strict they'd save out as the kind of Jpeg that Photoshop expects. 

In this case it's probably convenient to use a type that doesn't contain EXIF as standard. 

There is a similar problem when trying to parse Photoshop TIFF with Java.  Photoshop wants the extra area of metadata in the files. 

If you open a regular Jpeg and one of these from instagram in Hex you can see the different magicbytes after the "I'm a jpeg" bit. 

It's also the same situation if you save an image from instagram without using your plugin, just manually right-click->save as in browsers.  

Not your fault mate.

**Comment by nitriques** [comment]
Hahah thanks for the explanation.

Maybe a saveForAdobe switch could be added if we could find what needs to be changed in order for PS to open it! That would be sick.

**Comment by munki-boy** [comment]
This applies to all Adobe apps apart from probably Lightroom and Photoshop Camera Raw plugin.  You forgot saveForAdobe(TM) ha ha

**Comment by munki-boy** [comment]
JPEG image files begin with FF D8 and end with FF D9. JPEG/JFIF files contain the ASCII code for "JFIF" (4A 46 49 46) as a null terminated string. JPEG/Exif files contain the ASCII code for "Exif" (45 78 69 66) also as a null terminated string, followed by more metadata about the file.

**Comment by nitriques** [comment]
> saveForAdobe(TM) 

Yeah let's name it `saveForAdocrap`

Are those byte safes to blindly insert ?

**Comment by munki-boy** [comment]
Yes, it starts at the seventh byte but it didn't make the file ok for Photoshop when I tried just now. 

Photoshop is doing "Strict Typing" thought you'd approve of that?

**Comment by munki-boy** [comment]
Check this out: https://en.wikipedia.org/wiki/JPEG_File_Interchange_Format

**Comment by nitriques** [comment]
> Photoshop is doing "Strict Typing" thought you'd approve of that?

Normally, I would. But I hate Adobe so much, I can't say it when they've made the right choice ahaha

**Comment by munki-boy** [comment]
ha ha F\* the man!

**Comment by dbussiere** [comment]
Here's the trick :

https://imgur.com/gallery/ipHIV

**Comment by nitriques** [comment]
@dbussiere The link is down, wanna tell us more ?

**Comment by laocoi** [comment]
Goto Edit -> Preferences -> File Handling -> Camera Raw Preferences
In JPEG and TIFF handling part, choose option: Automatically open all supported...

**Comment by madranet** [comment]
> Goto Edit -> Preferences -> File Handling -> Camera Raw Preferences In JPEG and TIFF handling part, choose option: Automatically open all supported...

Thanks. This works but the downside is you now have to go through the extra Camera Raw import dialogue for *every* JPG and TIFF you open in Photoshop. So, if you only occasionally open an Instagram image in Photoshop,  it's probably less hassle to just open it first in some other image viewer [eg: Preview on OSX] and then export it again. Photoshop will open it no problem, after that.

BTW - I'm not sure why this has been submitted as a Shift+Click issue. It's a Photoshop problem. It happens even with photos downloaded from Instagram through source code diving.

**Comment by nitriques** [comment]
hey @madranet @munki-boy @laocoi 

We are pleased to announced that shift click is back ! We wrote the whole thing, but it looks and feel like the 2015 version :) Please re-install the extension https://shft.cl/chrome Happy sharing !
